torch/optim/sparse_adam.py

```diff
@@ -104,14 +104,16 @@ def step(self, closure=None):
     update the first moment, the second moment, and the parameter based on all values
     of the gradients, SparseAdam only updates the moments and parameters corresponding
     to the non-zero values of the gradients.
-    
+
     A simplified way of thinking about the implementation is as such:
+
     1. Create a mask of the non-zero values in the sparse gradients. For example,
-    if your gradient is [0, 5, 0, 0, 9], the mask would be [0, 1, 0, 0, 1].
-    2. Apply this mask over the running moments and do computation on only the
+    2. f your gradient is [0, 5, 0, 0, 9], the mask would be [0, 1, 0, 0, 1].
+    3. Apply this mask over the running moments and do computation on only the
+
```

Comment:
Sorry can I ask for clarification here. What you mention in this update, is exactly what I am doing. But the optimizer won't work in this case still, because the gradients themselves need to satisfy `param.grad.is_sparse()`. perhaps this could be added to clarify? Unless I'm missing something?

Reply:
The later comments hope to explain that part. I had wanted to motivate the reasoning of the implementation first, but I agree with moving the disclaimer up front.

Reply:
got it ! thanks!